feat: Add Multi-Factor Authentication (MFA) and Session Timeout

gauravanand809 opened the pull request:
**Add Multi-Factor Authentication (MFA) and Session Timeout**

1. Enabled Clerk’s built-in MFA to enhance user authentication security.

1. Implemented session timeout and auto-logout after a period of inactivity.

https://github.com/user-attachments/assets/5d4e465f-8c1f-497e-814f-ef7aa38a77b0

gauravanand809 committed:
feat: enhance Clerk session security and role-based access (closes #15)

gauravanand809 committed:
fix: replace Math.random with deterministic values for CI compatibility

gauravanand809 commented:
@amitkumardemo I think the issue#15 has been fixed, it would be great if you give a look and accept the merge

gauravanand809 committed:
chore: add labeler config to satisfy GitHub Actions

gauravanand809 committed:
Moved GenerateBackupCodes into its own component to resolve build-time type constraint violation

gauravanand809 commented:
@amitkumardemo Will you please look the trails and approve the solution, All quality checks and other passed but vercel need your authorization, hence do it get merged

amitkumardemo commented:
<img width="1520" height="688" alt="image" src="https://github.com/user-attachments/assets/aace66a3-a60b-4854-af8c-c4134730db4a" />
fix this issue

gauravanand809 committed:
header and footer path fixed

gauravanand809 committed:
header import fixed

gauravanand809 commented:
@amitkumardemo I have fixed the issue, I have changed the file name of header.js -> Header.js and it was not registered by git and hence can' t resolve the path, I have fixed it and tested it in vercel

If MFA creates an Issue, then do one thing, in your clerk account

- In the Clerk Dashboard, navigate to the [Multi-factor](https://dashboard.clerk.com/last-active?path=user-authentication/multi-factor) page.
- For the purpose of this guide, toggle on both the Authenticator application and Backup codes strategies.
- Select Save.

<img width="1885" height="885" alt="image" src="https://github.com/user-attachments/assets/69f2bb64-c31a-4ec7-990a-75b7a2ecb525" />

gauravanand809 committed:
fixed the dark mode issue of Authenticator app not detecting the qr code

gauravanand809 committed:
fixed the dark mode issue of Authenticator app not detecting the qr code

amitkumardemo commented:
[12:54:56.277] Running build in Washington, D.C., USA (East) – iad1
[12:54:56.278] Build machine configuration: 2 cores, 8 GB
[12:54:56.295] Cloning github.com/amitkumardemo/EdgeCareer (Branch: auth/clerk-enhanced-session, Commit: 08b4523)
[12:54:56.302] Skipping build cache, deployment was triggered without cache.
[12:54:56.699] Cloning completed: 404.000ms
[12:54:59.082] Running "vercel build"
[12:54:59.613] Vercel CLI 44.4.3
[12:54:59.938] Installing dependencies...
[12:55:22.463] 
[12:55:22.465] > ai-career-coach@0.1.0 postinstall
[12:55:22.466] > prisma generate
[12:55:22.466] 
[12:55:23.118] Prisma schema loaded from prisma/schema.prisma
[12:55:23.486] 
[12:55:23.487] ✔ Generated Prisma Client (v6.2.1) to ./node_modules/@prisma/client in 77ms
[12:55:23.488] 
[12:55:23.488] Start by importing your Prisma Client (See: https://pris.ly/d/importing-client)
[12:55:23.488] 
[12:55:23.488] Tip: Need your database queries to be 1000x faster? Accelerate offers you that and more: https://pris.ly/tip-2-accelerate
[12:55:23.489] 
[12:55:23.698] 
[12:55:23.698] added 695 packages in 24s
[12:55:23.699] 
[12:55:23.699] 275 packages are looking for funding
[12:55:23.699]   run `npm fund` for details
[12:55:23.769] Detected Next.js version: 15.2.3
[12:55:23.774] Running "npm run build"
[12:55:23.898] 
[12:55:23.898] > ai-career-coach@0.1.0 build
[12:55:23.898] > next build
[12:55:23.898] 
[12:55:24.552] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[12:55:24.553] This information is used to shape Next.js' roadmap and prioritize features.
[12:55:24.553] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[12:55:24.553] https://nextjs.org/telemetry
[12:55:24.553] 
[12:55:24.612]    ▲ Next.js 15.2.3
[12:55:24.613] 
[12:55:24.680]    Creating an optimized production build ...
[12:55:51.562] Failed to compile.
[12:55:51.563] 
[12:55:51.563] ./app/layout.jsx
[12:55:51.564] Module not found: Can't resolve '../components/Header'
[12:55:51.564] 
[12:55:51.564] https://nextjs.org/docs/messages/module-not-found
[12:55:51.564] 
[12:55:51.578] 
[12:55:51.579] > Build failed because of webpack errors
[12:55:51.613] Error: Command "npm run build" exited with 1
[12:55:51.952] 
[12:55:55.402] Exiting build container

gauravanand809 commented:
@amitkumardemo I have fixed this issue completely and deployed on my side as pasted the image, this is the past commit not the latest commit, check this out this is deployed: [edge-career.vercel.app](https://edge-career.vercel.app/)

gauravanand809 commented:
@amitkumardemo I shall open a new PR maybe something is not going well in this PR, hope you understand
